actionpack/lib/action_dispatch/middleware/host_authorization.rb

```diff
@@ -56,17 +60,45 @@ def sanitize_string(host)
         end
     end
 
-    DEFAULT_RESPONSE_APP = -> env do
-      request = Request.new(env)
+    class DefaultResponseApp # :nodoc:
+      RESPONSE_STATUS = 403
+
+      def call(env)
+        request = Request.new(env)
+        format = request.xhr? ? "text/plain" : "text/html"
+
+        if request.local? || request.get_header("action_dispatch.show_detailed_exceptions")
+          response(format, template_body(request))
+        else
+          log_error(request)
+          response(format)
+        end
+      end
+
+      private
+        def template_body(request)
+          template = DebugView.new(host: request.host)
+          template.render(template: "rescues/blocked_host", layout: "rescues/layout")
+        end
+
+        def response(format, body = "")
+          [RESPONSE_STATUS,
+           { "Content-Type" => "#{format}; charset=#{Response.default_charset}",
+             "Content-Length" => body.bytesize.to_s },
+           [body]]
+        end
+
+        def log_error(request)
+          logger = available_logger(request)
 
-      format = request.xhr? ? "text/plain" : "text/html"
-      template = DebugView.new(host: request.host)
-      body = template.render(template: "rescues/blocked_host", layout: "rescues/layout")
+          return unless logger
 
-      [403, {
-        "Content-Type" => "#{format}; charset=#{Response.default_charset}",
-        "Content-Length" => body.bytesize.to_s,
-      }, [body]]
+          logger.info("[#{self.class.name}] Blocked host: #{request.host}")
```

Comment:
I think this should use the `error` level, since it's also going to result in an error page for users.

Reply:
The user trying to access the app won't see the error either way (since you are returning a blank response, which is correct).

From the perspective of someone reading the app logs, I think this is something you really want to be aware of, hence suggesting error.

Reply:
Fixed